### PR TITLE
got 0.75 (new formula)

### PR DESCRIPTION
**Pull request by wolverian**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by wolverian** [comment]
Gah, I just noticed https://github.com/Homebrew/homebrew-core/pull/106492.

File: Formula/got.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Got < Formula
+  desc "Version control system"
+  homepage "https://gameoftrees.org"
+  url "https://gameoftrees.org/releases/portable/got-portable-0.75.tar.gz"
+  sha256 "b35324893c9aefbd6bf49602a6c9bb9d6e20670aff3c9734fefead0205bb9cb2"
+  license "ISC"
+
+  depends_on "ncurses" => :build
+  depends_on "openssl@3" => :build
+  depends_on "ossp-uuid" => :build
```
Comment (carlocab) [inline Formula/got.rb:10]: These are likely needed at runtime. Does macOS `ncurses` not work?

File: Formula/got.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  uses_from_macos "bison" => :build
+
+  def install
+    system "./configure", *std_configure_args, "--disable-silent-rules"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/got", "--version"
```
Comment (carlocab) [inline Formula/got.rb:20]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/got.rb (hunk 1 above)
Comment (wolverian) [reply Formula/got.rb:10]: > These are likely needed at runtime.

Yep, brainfart on my part.

> Does macOS ncurses not work?

`uses_from_macos "ncurses"` doesn't work. The configure script wants `ncursesw` and `panelw`, which I think are not provided by macOS ncurses.

File: Formula/got.rb (hunk 2 above)
Comment (wolverian) [reply Formula/got.rb:20]: I borrowed the test implementation from https://github.com/Homebrew/homebrew-core/pull/106492.

File: Formula/got.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "https://gameoftrees.org"
+  url "https://gameoftrees.org/releases/portable/got-portable-0.75.tar.gz"
+  sha256 "b35324893c9aefbd6bf49602a6c9bb9d6e20670aff3c9734fefead0205bb9cb2"
+  license "ISC"
+
+  depends_on "ncurses"
+  depends_on "openssl@3"
+  depends_on "ossp-uuid"
+
+  uses_from_macos "bison" => :build
```
Comment (carlocab) [inline Formula/got.rb:12]: Looks like this needs to be `depends_on` as well.

Comment (wolverian) [reply Formula/got.rb:12]: How do you see that? Asking to learn. 🙂

Comment (carlocab) [reply Formula/got.rb:12]: That's a guess based on the CI failure here: https://github.com/Homebrew/homebrew-core/actions/runs/3062430582/jobs/4943465511#step:6:189

**Comment by carlocab** [comment]
Linux may also need `libbsd`, `libmd`, and `util-linux`. But CI will tell us shortly.

File: Formula/got.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://gameoftrees.org/releases/portable/got-portable-0.75.tar.gz"
+  sha256 "b35324893c9aefbd6bf49602a6c9bb9d6e20670aff3c9734fefead0205bb9cb2"
+  license "ISC"
+
+  depends_on "bison" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libevent"
+  depends_on "ncurses"
+  depends_on "openssl@1.1"
+  depends_on "ossp-uuid"
```
Comment (carlocab) [inline Formula/got.rb:13]:
```suggestion
  depends_on "ossp-uuid"
  uses_from_macos "zlib"
```

File: Formula/got.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  uses_from_macos "zlib"
+
+  def install
```
Comment (carlocab) [inline Formula/got.rb:16]:
```suggestion
  uses_from_macos "zlib"

  on_linux do
    depends_on "libmd"
  end

  def install
```

File: Formula/got.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  depends_on "bison" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libevent"
+  depends_on "ncurses"
+  depends_on "openssl@1.1"
+  depends_on "ossp-uuid"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "libmd"
```
Comment (carlocab) [inline Formula/got.rb:17]:
```suggestion
    depends_on "libmd"
    depends_on "util-linux" # for uuid.h
```

File: Formula/got.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  desc "Version control system"
+  homepage "https://gameoftrees.org"
+  url "https://gameoftrees.org/releases/portable/got-portable-0.75.tar.gz"
+  sha256 "b35324893c9aefbd6bf49602a6c9bb9d6e20670aff3c9734fefead0205bb9cb2"
+  license "ISC"
+
+  depends_on "bison" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libevent"
+  depends_on :macos # FIXME: ossp-uuid and util-linux conflict with each other.
```
Comment (carlocab) [inline Formula/got.rb:11]: Ok, I don't know how to get this to work on Linux. We can ship this on macOS for now.

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**